internal/types/host/keepalived_host_collector_host.go

```diff
@@ -0,0 +1,270 @@
+			return syscall.SIGUSR2
+		default:
+			logrus.WithField("signal", sigString).Fatal("Unsupported signal for your keepalived")
+		}
+	}
+
+	sigNumCommand := "keepalived --signum=" + sigString
+	cmd := exec.Command("bash", "-c", sigNumCommand)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	if err := cmd.Run(); err != nil {
+		logrus.WithFields(logrus.Fields{"signal": sigString, "stderr": stderr.String()}).WithError(err).Fatal("Error getting signum")
+	}
+
+	signum, err := strconv.ParseInt(stdout.String(), 10, 32)
+	if err != nil {
+		logrus.WithFields(logrus.Fields{"signal": sigString, "signum": stdout.String()}).WithError(err).Fatal("Error parsing signum result")
+	}
+
+	return syscall.Signal(signum)
+}
+
+func (k *KeepalivedHostCollectorHost) HasSigNumSupport() bool {
+	return k.version == nil || k.version.GreaterThanOrEqual(sigNumSupportedVersion)
+}
+
+func (k *KeepalivedHostCollectorHost) jsonVrrps() ([]collector.VRRP, error) {
+	err := k.signal(k.SIGJSON)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to send JSON signal to keepalived")
+		return nil, err
+	}
+
+	f, err := os.Open("/tmp/keepalived.json")
+	if err != nil {
+		logrus.WithError(err).Error("Failed to open /tmp/keepalived.json")
+		return nil, err
+	}
+	defer f.Close()
+
+	VRRPs, err := collector.ParseJSON(f)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to decode keepalived.json to VRRPStats array structure")
+		return nil, err
+	}
+
+	return VRRPs, nil
+}
+
+func (k *KeepalivedHostCollectorHost) statsVrrps() (map[string]*collector.VRRPStats, error) {
+	err := k.signal(k.SIGSTATS)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to send STATS signal to keepalived")
+		return nil, err
+	}
+
+	f, err := os.Open("/tmp/keepalived.stats")
+	if err != nil {
+		logrus.WithError(err).Error("Failed to open /tmp/keepalived.stats")
+		return nil, err
+	}
+	defer f.Close()
+
+	vrrpStats, err := collector.ParseStats(f)
+	if err != nil {
+		return nil, err
+	}
+
+	return vrrpStats, nil
+}
+
+func (k *KeepalivedHostCollectorHost) dataVrrps() (map[string]*collector.VRRPData, error) {
+	err := k.signal(k.SIGDATA)
+	if err != nil {
+		logrus.WithError(err).Error("Failed to send DATA signal to keepalived")
+		return nil, err
+	}
+
+	f, err := os.Open("/tmp/keepalived.data")
+	if err != nil {
+		logrus.WithError(err).Error("Failed to open /tmp/keepalived.data")
+		return nil, err
+	}
+	defer f.Close()
+
+	vrrpData, err := collector.ParseVRRPData(f)
+	if err != nil {
+		return nil, err
+	}
+
+	return vrrpData, nil
+}
+
+func (k *KeepalivedHostCollectorHost) scriptVrrps() ([]collector.VRRPScript, error) {
+	f, err := os.Open("/tmp/keepalived.data")
+	if err != nil {
+		logrus.WithError(err).Error("Failed to open /tmp/keepalived.data")
+		return nil, err
+	}
+	defer f.Close()
+
+	return collector.ParseVRRPScript(f), nil
+}
+
+func (k *KeepalivedHostCollectorHost) GetKeepalivedStats() (*collector.KeepalivedStats, error) {
```

Comment:
Doesn't this function need to be duplicated for all collectors? :grimacing:

Reply:
I'll try to make some changes.